backend/__init__.py

```diff
@@ -835,23 +838,50 @@ def notification_runner():
                 for url in n_object['notification_urls']:
                     apobj.add(url.strip())
 
-                n_body = n_object['watch_url']
+                # Get the notification body from datastore
+                n_body = datastore.data['settings']['application']['notification_body']
+                # Get the notification title from the datastore
+                n_title = datastore.data['settings']['application']['notification_title']
+
+                # Insert variables into the notification content
+                notification_parameters = create_notification_parameters(n_object)
+                raw_notification_text = [n_body, n_title]
 
-                # 65 - Append URL of instance to the notification if it is set.
-                base_url = os.getenv('BASE_URL')
-                if base_url != None:
-                    n_body += "\n" + base_url
+                parameterised_notification_text = dict(
+                    [
+                        (i, n.replace(n, n.format(**notification_parameters)))
+                        for i, n in zip(['body', 'title'], raw_notification_text)
```

Comment:
what does this do?

Reply:
`dict.items()` returns a list of tuples, with each tuple being a key,value pair in the `dict`. We can reverse engineer this and create a dict by giving a list of tuples to `dict()`. I suppose the list is optional and we could use dict comprehension, but I find this syntax clearer.

`n.replace(n, n.format(**notification_parameters))` is

- taking the user's title and body (with the tokens)
- deconstructing the `notification_parameters` dictionary into key, value pairs
- applying `.format()` to the user strings with these key, value pairs 

any tokens not in the string are discarded by `format()` so it's safe.